vllm/model_executor/layers/fused_moe/utils.py

```diff
@@ -98,6 +99,16 @@ def _resize_cache(x: torch.Tensor, v: tuple[int, ...]) -> torch.Tensor:
     return x.flatten()[:prod(v)].view(*v)
 
 
+def _fp4_quantize(
+    A: torch.Tensor,
+    A_scale: Optional[torch.Tensor],
+    is_sf_swizzled_layout: bool,
+) -> tuple[torch.Tensor]:
+    return fp4_quantize(A,
+                        A_scale,
+                        is_sf_swizzled_layout=is_sf_swizzled_layout)
```

Comment:
Is there a reason why we need to use flashinfer's quantize function over ours?

Reply:
We need to quantize without the scaling factor being swizzled since the communication op is scheduled after quantization. The pattern is quantize(no_swizzle_sf) -> all-gather -> swizzle_sf. vllm's native scaled_fp4_quant cannot do so.